bgstally/bgstally.py

```diff
@@ -125,23 +126,20 @@ def journal_entry(self, cmdr, is_beta, system, station, entry, state):
             self.debug.logger.error(f"The EDMC Version is too old, please upgrade to v5.6.0 or later", exc_info=e)
             return
 
-        activity: Activity = self.activity_manager.get_current_activity()
-
         # Total hack for now. We need cmdr in Activity to allow us to send it to the API when the user changes values in the UI.
         # What **should** happen is each Activity object should be associated with a single CMDR, and then all reporting
         # kept separate per CMDR.
+        activity:Activity = self.activity_manager.get_current_activity()
         activity.cmdr = cmdr
 
-        dirty: bool = False
-
-        if entry.get('event') in ['StartUp', 'Location', 'FSDJump', 'CarrierJump']:
-            activity.system_entered(entry, self.state)
-            self.colonisation.journal_entry(cmdr, is_beta, system, station, entry, state)
-            dirty = True
-
         mission:dict = self.mission_log.get_mission(entry.get('MissionID'))
 
         match entry.get('event'):
+            case 'StartUp' | 'Location' | 'FSDJump' | 'CarrierJump':
```

Comment:
This is bad.  The reason this code is in a separate block above is so that it **ALWAYS** gets executed for these event types.

The python `match` `case` here will only execute the first successful match, and then exit. So, the `case 'Location' | 'StartUp'` lower down will never be executed, which will cause problems as the `station_faction` in that code block will never be set.